core/controllers/acl_decorators.py

```diff
@@ -1297,9 +1297,6 @@ def test_can_access(self, thread_id, **kwargs):
             UnauthorizedUserException. The user does not have credentials to
                 view an exploration feedback.
         """
-        if '.' not in thread_id:
```

Comment:
Shouldn't it be more like "check that thread_id matches given regex"? This seems like a very loose check.

Reply:
I want to confirm that -- Schema validation process is done before reaching acl layer. 
The thread_id reaching through this decorator via API must be a valid one.
I have added backend test similar to feedback comment thread decorator in order to pass coverage.

Reply:
OK, I'm a bit confused then. If we validate this as part of schema before reaching the ACL layer, then why do we need this check here?

Also, it is correct that the schema validation process happens fully before reaching ACL layer, right? I'm not sure if you're stating this or asking whether it should be the case. If the latter then I'm totally fine with it.

Reply:
>If we validate this as part of schema before reaching the ACL layer, then why do we need this check here?

These decorators are used by 3 handlers in which I have added schema, but I am wondering of the case when any of the new handlers will use these decorator & due to any reason they do not write additional validators for thread_id in schema, to check via its regex. Thus I think this check should be preserved.

>Also, it is correct that the schema validation process happens fully before reaching ACL layer, right?

True, but for those handlers in which schemas are written.

By above statement I only want to keep this check preserved, but I am looking for your suggestions.

Reply:
I see. I think we are ultimately going to have every handler have schema validations, right? So maybe what you can do here is to have the full regex validation but also have a TODO above it saying "remove this after all handlers have full schema validation". What do you think?

Reply:
Added regex but before adding a TODO I want  to confirm that I am assuming the condition when any contributor miss to add regex in the schema so for that condition we will miss a tight check for thread id.
Here we have schema but it is not tight.
So I believe this regex check should be preserved here.
As far as these decorator are concerned -- this PR contains schemas for the handlers which uses these decorators.
So currently they do not depend on addition of schema for remaining handlers.

Reply:
OK, that seems reasonable to me too, thanks for thinking about it. I think what you say sounds fine and you can add a comment like "This should already be checked by the controller handler argument schemas, but we are adding it here for additional safety." so that future developers don't remove it.

Reply:
Thank you @seanlip, I have added the comment, please take a look.
Thank you.